notebooks/122-quantizing-model-with-accuracy-control/122-speech-recognition-quantization-wav2vec2.ipynb

```diff
@@ -0,0 +1,712 @@
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sys\n",
+    "import tarfile\n",
+    "import torch\n",
+    "\n",
+    "from transformers import Wav2Vec2ForCTC\n",
+    "\n",
+    "sys.path.append(\"../utils\")\n",
+    "from notebook_utils import download_file"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<a id=\"2\"></a>\n",
+    "## Settings [&#8657;](#0)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "\n",
+    "# Set the data and model directories, model source URL and model filename.\n",
+    "MODEL_DIR = Path(\"model\")\n",
+    "DATA_DIR = Path(\"../data/datasets/librispeech\")\n",
+    "MODEL_DIR.mkdir(exist_ok=True)\n",
+    "DATA_DIR.mkdir(exist_ok=True)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<a id=\"3\"></a>\n",
+    "## Prepare the Model [&#8657;](#0)\n",
+    "Perform the following:\n",
+    "- Download and unpack a pre-trained Wav2Vec2 model.\n",
+    "- Run model conversion API to convert the model to the OpenVINO Intermediate Representation (OpenVINO IR)."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "download_file(\"https://huggingface.co/facebook/wav2vec2-base-960h/resolve/main/pytorch_model.bin\", directory=Path(MODEL_DIR) / 'pytorch', show_progress=True)\n",
+    "download_file(\"https://huggingface.co/facebook/wav2vec2-base-960h/resolve/main/config.json\", directory=Path(MODEL_DIR) / 'pytorch', show_progress=False)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Import all dependencies to load the original PyTorch model and convert it to the OpenVINO Intermediate Representation (OpenVINO IR).."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "BATCH_SIZE = 1\n",
+    "MAX_SEQ_LENGTH = 30480\n",
+    "\n",
+    "\n",
+    "torch_model = Wav2Vec2ForCTC.from_pretrained(Path(MODEL_DIR) / 'pytorch')"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import openvino\n",
+    "\n",
+    "\n",
+    "default_input = torch.zeros([1, MAX_SEQ_LENGTH], dtype=torch.float)\n",
+    "ov_model = openvino.convert_model(torch_model, example_input=default_input)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "<a id=\"4\"></a>\n",
```

Comment:
I am really concerned about creating own objects to work with datasets and metrics while everything is already available in public. I don't think that these how people go about this when they train/validate models.
LibriSpeech is available both in [torchaudio](https://pytorch.org/audio/main/generated/torchaudio.datasets.LIBRISPEECH.html) and HuggingFace [datasets](https://huggingface.co/datasets/librispeech_asr) library. Word Error rate is available in [torchmetrics](https://pytorch.org/torcheval/main/generated/torcheval.metrics.WordErrorRate.html) as well. 
You are doing a role modeling here for the users. I don't think they will be happy to write everything from scratch just to optimize the model.

Reply:
Redesigned. Now it is used a lightweight huggingface dataset `librispeech_asr_dummy` with preprocessing function, WordErrorRate from torchmetrics, Wav2Vec2Processor for pre- and post-processing steps and no direct file downloads.

Reply:
Thanks, now it makes sense.